muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/records/body/application/matriculation-eligibility-row/matriculation-eligibility-row.tsx

```diff
@@ -0,0 +1,174 @@
+import * as React from "react";
+import { connect } from "react-redux";
+import { i18nType } from "~/reducers/base/i18n";
+import { Dispatch } from "redux";
+import { StateType } from '~/reducers';
+import mApi from '~/lib/mApi';
+
+/**
+ * Enum describing matriculation eligibility
+ * 
+ * @author Antti Leppä <antti.leppa@metatavu.fi>
+ */
+enum EligbleEnum { 
+  TRUE, 
+  FALSE,
+  UNKNOWN
+};
+
+/**
+ * Interface representing matriculation eligibility REST model 
+ * 
+ * @author Antti Leppä <antti.leppa@metatavu.fi>
+ */
+interface MatriculationEligibilityType {
+  eligible: boolean;
+  requirePassingGrades: number;
+  acceptedCourseCount: number;
+  acceptedTransferCreditCount: number;
+}
+
+/**
+ * Interface representing MatriculationEligibilityRow component properties
+ * 
+ * @author Heikki Kurhinen <heikki.kurhinen@metatavu.fi>
+ */
+interface MatriculationEligibilityRowProps {
+  code: string,
+  subjectCode: string,
+  i18n: i18nType
+}
+
+/**
+ * Interface representing MatriculationEligibilityRow component state
+ * 
+ * @author Heikki Kurhinen <heikki.kurhinen@metatavu.fi>
+ */
+interface MatriculationEligibilityRowState {
+  eligible: EligbleEnum,
+  requiredCount: number;
+  acceptedCount: number;
+  loading: boolean
+}
+
+/**
+ * MatriculationEligibilityRow component
+ * 
+ * @author Heikki Kurhinen <heikki.kurhinen@metatavu.fi>
+ */
+class MatriculationEligibilityRow extends React.Component<MatriculationEligibilityRowProps, MatriculationEligibilityRowState> {
+  constructor(props: MatriculationEligibilityRowProps){
+    super(props);
+    
+    this.state = {
+      eligible: EligbleEnum.UNKNOWN,
+      requiredCount: 0,
+      acceptedCount: 0,
+      loading: false
+    }
+  }
+
+  /**
+   * Component did mount life-cycle method  
+   * 
+   * Reads available matriculation subjects from REST API
+   */
+  componentDidMount() {
+    if (!this.state.loading) {
```

Comment:
componentDidMount gets called once, instead of initializing loading at false, initialize it at true; so the component initializes at loading rather than setting the state later, and perform the loading during the componentdidmount.

However a more efficient way would be to use the componentWillMount function so that if the mApi manages to call quickly the initial render gets called without loading (it's to doubt that would be the case).

Reply:
Moved to componentWillMount and changed loading to be initialized as true